### PR TITLE
feat(spdk): updating to SPDK 23.01

### PR DESCRIPTION
**Pull request by dsavitskiy**
* Bdev API can only accessed from SPDK thread
* New fields in various SPDK structures
* New AIO Bdev options
* New malloc Bdev options
* Async Qpair connect and subsystem events are made optional features

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: shell.nix [hunk 1]
```diff
@@ -47,6 +47,9 @@ mkShell {
     utillinux
     gnuplot
     libunwind
+    autoconf
+    automake
+    yasm
```
Comment (tiagolobocastro) [inline shell.nix:52]: not related, but at some point it'd be nice to make one shell nix depend on the other

**Review by tiagolobocastro — APPROVED** [review]

File: shell.nix (hunk 1 above)
Comment (dsavitskiy) [reply shell.nix:52]: we actually need only one, "ci.nix". The default one "shell.nix" is for using system's Rust, which is not a good idea is general.  So I want to remove shell.nix completely and make it alias to ci.nix.
As for ability to build with the system's Rust, I'm planning to do it with the shell arg we already have: "--arg norust true".

**Review by dsharma-dc — APPROVED** [review]

**Review by mtzaurus — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors merge

**Comment by dsavitskiy** [comment]
bors try

**Comment by tiagolobocastro** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**